crates/sui-core/src/transaction_input_checker.rs

```diff
@@ -90,7 +91,7 @@ where
 #[instrument(level = "trace", skip_all)]
 async fn check_gas<S>(
     store: &SuiDataStore<S>,
-    gas_payment_id: ObjectID,
+    gas_payment: &ObjectRef,
```

Comment:
Technically the digest isn't necessary and we can just pass the ID and version here.